web/documentserver-example/csharp-mvc/WebEditor.ashx.cs

```diff
@@ -812,6 +812,26 @@ private static void Reference(HttpContext context)
                 }
             }
 
+            if (fileName == "" && body.ContainsKey("link"))
+            {
+                if (!body["link"].ToString().Contains(DocManagerHelper.GetServerUrl(false)))
+                {
+                    context.Response.Write(jss.Serialize(new Dictionary<string, string>() {
+                        { "url", body["link"].ToString() },
+                        { "directUrl", body["link"].ToString() }
+                    }));
+                    return;
+                }
+
+                Uri linkUri = new Uri(body["link"].ToString());
+                fileName = HttpUtility.ParseQueryString(linkUri.Query).Get("fileName");
+                if (string.IsNullOrEmpty(fileName) || !File.Exists(DocManagerHelper.StoragePath(fileName, null)))
+                {
+                    context.Response.Write("{ \"error\": \"File is not exist\"}");
+                    return;
+                }
```

Comment:
You access the link by the key from the body and convert it into the string 4 times. I suggest you assign the link's value to the local variable and then use it.